### ISSUE TITLE
Zero-shot Arabidopsis ovule benchmark (Fig. 3C): image IDs and inference settings

### ISSUE THREAD

**Issue by postnubilaphoebus** [yu-lab-vt/FOCUS-3D#10]
Hi, great work on FOCUS-3D. Thanks for open-sourcing it.

I'm trying to reproduce the zero-shot Arabidopsis ovule result in Fig. 3C (mean AP@0.5 = 0.830) and couldn't find a few details in the paper or repo:

1. Which three images from the ovule dataset (Vijayan et al. 2024 / BioImage Archive S-BIAD102645) were used? We're testing with samples 1136, 1137, and 1139 and want to confirm these are the same ones.
2. The Methods note that "the axial resolution ratio and cell-radius parameter were adjusted according to the imaging resolution and apparent cell size of each dataset." What values were used for this dataset, and how was "apparent cell size" measured?
3. Which released checkpoint (general, membrane, or nuclei) corresponds to what was evaluated for this benchmark?

Would appreciate any pointers. Happy to share our reproduction setup if useful. Thanks!

**Comment by Qinghua24** [comment]
Hi,
Thank you for your interest in FOCUS-3D.
1. We used three images—1136, 1139, and 1170—for our evaluation.
2. We used a default radius of 15 for the evaluation. Regarding the choice of cell radius, we typically select cells of moderate size within the field of view and measure their radius in pixels; as long as the radius estimate is reasonably accurate, the results will not vary significantly. Axial resolution has little impact on the segmentation results (we used 5.0).
3. We used an intermediate checkpoint for the evaluation, which did not include S-BIAD1026 in the training set. The version ultimately released to users incorporates this dataset; therefore, using the "general/nuclei" model for evaluation is likely to yield better results than those shown in Fig. 3C.

Qinghua

**Comment by postnubilaphoebus** [comment]
Hi, Thanks for your pointers, I am following up with what I found.

I ran FOCUS-3D zero-shot on 1136/1137/1139 (BioImage Archive S-BIAD102645), membrane checkpoint, cell_radius measured from each image's own ground truth (~31px), score threshold 0.7 and mask threshold 0.5, paper's stated default settings otherwise. Scored against `_cellseg.tif` with Hungarian-matched instance IoU: AP@0.5 of 0.28, 0.72, 0.70 (mean 0.57). General checkpoint with cell_radius=15 on 1136/1139/1170 gave 0.29, 0.74, 0.62 (mean 0.55). Both well below the reported 0.830, and 1136 is consistently the worst.

On 1136 I also see large tile-shaped instances rather than cell-shaped ones. The biggest is 14.6M voxels with a bounding box exactly on the patch/stride grid (starts at Y=128=2x64, X=256=4x64, Z-extent=224=7x32) and dimmer than surrounding tissue at the same depth. Given the block-stitching method (face-based ID matching across blocks, no size check mentioned in the within-block confidence x mask-quality filtering), this looks like a low-contrast block producing a near-block-filling mask that then chains across neighbors. Is this a known failure mode, and is there a setting to suppress it?

Am I missing a parameter or preprocessing step? Happy to share scripts/settings.

Best,
Laurids

<img width="858" height="475" alt="Image" src="https://github.com/user-attachments/assets/b8072f50-4efb-4dcb-8ef9-972b09707b33" />

**Comment by Qinghua24** [comment]
Hi, 

First, I would like to clarify that the evaluation in Fig. 3C utilized the nuclei channel of S-BIAD1026 rather than the membrane channel; this approach maintains consistency with the evaluations previously conducted on zebrafish, Drosophila, and C. elegans.

Thank you for raising the issue regarding membrane segmentation. We have previously observed this phenomenon when FOCUS-3D processes cells with very large radius or specific background conditions. This occurs because FOCUS-3D processes images in blocks; when a single block contains a uniform internal intensity, the model struggles to distinguish between foreground and background. Consequently, selecting an appropriate radius is crucial. For this specific case, we recommend setting a larger radius—such as 50 (note that near z=50, some cells actually exceed a radius of 50). Additionally, this issue can be mitigated by adjusting the `background_threshold` and `size_filter_max_size` parameters. The attached image shows the results I obtained; while not perfect, they are usable.

<img width="1984" height="1111" alt="Image" src="https://github.com/user-attachments/assets/d58745bf-25f3-44bb-9631-2dec88892c53" />

We are actively working on a definitive solution to this problem and will release an updated version soon. If you wish to discuss this further, please contact zhangqh24@mails.tsinghua.edu.cn. I am willing to provide any assistance.

Thank you again for your feedback!


Qinghua
